crypto/test/tests.go

```diff
@@ -32,3 +32,25 @@ func RandomSignerRFC6979(tb testing.TB) user.Signer {
 
 	return user.NewAutoIDSignerRFC6979(p.PrivateKey)
 }
+
+// SignedComponent describes component which can signed and the signature may be verified.
+type SignedComponent interface {
+	SignedData() []byte
+	Sign(user.Signer) error
+	VerifySignature() bool
+}
+
+// SignedDataComponent tests [SignedComponent] for valid data generation by SignedData function.
+func SignedDataComponent(tb testing.TB, signer user.Signer, cmp SignedComponent) {
```

Comment:
`signer` may be of type `neofscrypto.Signer`, isn't it?

Reply:
No, it can't be. If it `neofscrypto.Signer` we cover just couple of structs

Reply:
it's not about structs, it's about function body. It deals with `user.Signer` as `neofscrypto.Signer`